src/groovy/org/pih/warehouse/api/StockMovement.groovy

```diff
@@ -482,4 +482,13 @@ class StockMovement {
         def g = ApplicationHolder.application.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
         return "${g.message(code: 'enum.' + status?.getClass()?.getSimpleName() + '.' + status)}"
     }
+
+    Boolean canUserRollbackApproval(User user) {
+        // Approval can be rolled back by user who is approver or requestor
+        return approvers?.contains(user) || requestedBy == user
```

Comment:
I don't remember how groovy will handles that type of comparison, but maybe you've checked that - I am not sure if `requestedBy == user` will evaluate to `true` if we have different references? 
If it doesn't, then compare by id. Or maybe we have the same reference at this point?